extension-compose/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -10,6 +9,52 @@ import com.mapbox.maps.MapInitOptions
 import com.mapbox.maps.MapView
 import com.mapbox.maps.MapboxExperimental
 import com.mapbox.maps.MapboxMap
+import com.mapbox.maps.extension.compose.internal.MapApplier
+import com.mapbox.maps.extension.compose.internal.MapNode
+import com.mapbox.maps.extension.compose.internal.MapPreviewPlaceHolder
+import com.mapbox.maps.plugin.gestures.OnMapClickListener
+import com.mapbox.maps.plugin.gestures.OnMapLongClickListener
+import com.mapbox.maps.plugin.gestures.gestures
+import kotlinx.coroutines.awaitCancellation
+
+/**
+ * MapboxMapNode to observe/update the input arguments of MapboxMap.
+ */
+private class MapboxMapNode(
+  val controller: MapView,
+  var clickListener: OnMapClickListener?,
+  var longClickListener: OnMapLongClickListener?
+) : MapNode {
+  override fun onAttached() {
+    controller.gestures.apply {
+      clickListener?.let {
+        addOnMapClickListener(it)
+      }
+      longClickListener?.let {
+        addOnMapLongClickListener(it)
+      }
+    }
+  }
+
+  override fun onRemoved() {
+    cleanUp()
+  }
+
+  override fun onClear() {
+    cleanUp()
+  }
+
+  private fun cleanUp() {
+    controller.gestures.apply {
+      clickListener?.let {
+        removeOnMapClickListener(it)
+      }
```

Comment:
maybe set listener to `null`?

Reply:
the MapboxMapNode#clickListener is set to listener below, whether it's null or not.